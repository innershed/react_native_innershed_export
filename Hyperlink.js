import React from 'react';
import { View } from 'react-native';

import { Linking } from 'expo';

import PropTypes from 'prop-types';

import Touchable from './Touchable';

export default class Hyperlink extends React.Component {
    static propTypes = {
        href: PropTypes.string.isRequired,
        children: PropTypes.node,
        style: PropTypes.any
    };

    constructor(props) {
        super(props);

        this.open = async function() {
            const { href } = this.props;

            if (!await Linking.canOpenURL(href))
                return;
            else
                await Linking.openURL(href);
        }.bind(this);
    }

    render() {
        return (
            <Touchable onPress={this.open}>
                <View>
                    {this.props.children}
                </View>
            </Touchable>
        );
    }
}
