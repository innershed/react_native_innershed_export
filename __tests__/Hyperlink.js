import React from 'react';
import { Text } from 'react-native';

import renderer from 'react-test-renderer';

import Hyperlink from '../Hyperlink';

it('renders', () => {
    expect(renderer.create(
        <Hyperlink href={'https://www.innershed.com'}>
            <Text>https://www.innershed.com</Text>
        </Hyperlink>
    )).toMatchSnapshot();
});
