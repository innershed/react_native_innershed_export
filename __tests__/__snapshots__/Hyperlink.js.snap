// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={null}
  >
    <Text>
      https://www.innershed.com
    </Text>
  </View>
</View>
`;
